polars/polars-core/src/frame/asof_join/asof.rs

```diff
@@ -181,6 +182,57 @@ pub(super) fn join_asof_backward<T: PartialOrd + Copy + Debug>(
     out
 }
 
+pub(super) fn join_asof_nearest<T: PartialOrd + Copy + Debug + Sub<Output = T> + Bounded>(
+    left: &[T],
+    right: &[T],
+) -> Vec<Option<IdxSize>> {
+    let mut out = Vec::with_capacity(left.len());
+    let mut offset = 0 as IdxSize;
+    let max_value = <T as num_traits::Bounded>::max_value();
+    let mut dist: T = max_value;
+
+    for &val_l in left {
+        loop {
+            match right.get(offset as usize) {
+                Some(&val_r) => {
+                    // This is (val_r - val_l).abs(), but works on strings/dates
+                    let dist_curr = if val_r > val_l {
+                        val_r - val_l
+                    } else {
+                        val_l - val_r
+                    };
+                    if dist_curr <= dist {
+                        // candidate for match
+                        dist = dist_curr;
+                        offset += 1;
+                    } else {
+                        // distance has increased, we're now farther away, so previous element was closest
+                        out.push(Some(offset - 1));
+
+                        // reset distance
+                        dist = max_value;
+
+                        // The next left-item may match on the same item, so we need to rewind the offset
+                        offset -= 1;
+                        break;
+                    }
+                }
+
+                None => {
+                    // We've reached the end with no matches, so the last item was the nearest for all remaining
+                    out.extend(
+                        std::iter::repeat(if offset > 1 { Some(offset - 1) } else { None })
```

Comment:
It is faster to branch and then repeat.

Now you are doing the compare in the loop.

So :

```rust

if offset > 1 {

extend -> repeat Some

else {

extend -> repeat None

}
```

Reply:
Hmm, so actually I have this in there in case the right frame is empty. If match is `nearest` then the output will never have `None` values in it. I'm trying to see if there's an earlier check for an empty right-frame, in which case we can remove the `if` entirely.

Reply:
Ok, the branch is needed. Fixing now.